realm/realm-library/src/main/java/io/realm/CompactOnLaunchCallback.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
```

Comment:
I think the primary reason for Transaction to be an inner class is because we have two, one for DynamicRealm and one for Realm. 

You could argue it could have been an outer class like: 
```
public interface Transaction<T extends BaseRealm> {
   void execute(T realm);
}
```
But that would have exposed the concept of the `BaseRealm` in the public API.
But generally, I would tend to agree that outer classes look slightly better.

Inner classes have the benefit that they scope the interface to a very specific use case though.

IMO, In this case, it seems most natural to keep it as a separate interface